lib/ansible/modules/cloud/amazon/ecs_taskdefinition_facts.py

```diff
@@ -0,0 +1,346 @@
+            returned: when present
+            type: bool
+        sourcePath:
+            description: The path on the host container instance that is presented to the container.
+            returned: when present
+            type: string
+status:
+    description: The status of the task definition
+    returned: always
+    type: string
+requiresAttributes:
+    description: The container instance attributes required by your task
+    returned: when present
+    type: complex
+    contains:
+        name:
+            description: The name of the attribute.
+            returned: when present
+            type: string
+        value:
+            description: The value of the attribute.
+            returned: when present
+            type: string
+        targetType:
+            description: The type of the target with which to attach the attribute.
+            returned: when present
+            type: string
+        targetId:
+            description: The ID of the target.
+            returned: when present
+            type: string
+placementConstraints:
+    description: A list of placement constraint objects to use for tasks
+    returned: always
+    type: complex
+    contains:
+        type:
+            description: The type of constraint.
+            returned: when present
+            type: string
+        expression:
+            description: A cluster query language expression to apply to the constraint.
+            returned: when present
+            type: string
+'''  # NOQA
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+
+class EcsTaskManager:
+    """Handles ECS Tasks"""
+
+    def __init__(self, module):
+        self.module = module
+
+        try:
+            region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+            if not region:
+                module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION environment variables or in boto configuration file")
+            self.ecs = boto3_conn(module, conn_type='client', resource='ecs', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+        except boto.exception.NoAuthHandlerFound as e:
+            self.module.fail_json(msg="Can't authorize connection - %s" % str(e))
+
+    def describe_task_definitions(self, family):
+        # Return the full descriptions of the task definition
+        return self.ecs.describe_task_definition(taskDefinition=family)['taskDefinition']
+
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        task_definition=dict(required=True, type='str')
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    if not HAS_BOTO:
```

Comment:
Thanks for the contribution! Since new AWS modules are required to just use boto3, you can remove this. Your exception handling can also be updated to be for boto3. Each boto3 call should be in a try/except. There's some helpful info here: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md

Reply:
Hi, I've removed the boto dependency, leaving only boto3. After a bunch of horrible mistakes on the documentation syntax, the build seems to be stable now. Do you want me to squash my commits or should I leave it like that?